### ISSUE TITLE
Array to string on exception..

### ISSUE THREAD

**Issue by gmponos** [php-http/discovery#131]
| Q            | A
| ------------ | ---
| Bug?         | yes
| Version      | master

Hello can you check the following build: https://travis-ci.org/florianv/exchanger/jobs/474973917

The reason is that on this line: https://github.com/php-http/discovery/blob/master/src/Exception/NoCandidateFoundException.php#L27 variable $classes contains:

```
array (
  0 => 'Http\\Adapter\\Guzzle6\\Client',
  1 => 'Http\\Adapter\\Guzzle5\\Client',
  2 => 'Http\\Client\\Curl\\Client',
  3 => 'Http\\Client\\Socket\\Client',
  4 => 'Http\\Adapter\\Buzz\\Client',
  5 => 'Http\\Adapter\\React\\Client',
  6 => 'Http\\Adapter\\Cake\\Client',
  7 => 'Http\\Adapter\\Zend\\Client',
  8 => 'Http\\Adapter\\Artax\\Client',
  9 => 
  array (
    0 => 'Http\\Discovery\\Strategy\\CommonClassesStrategy',
    1 => 'buzzInstantiate',
  ),
)
```

You can see the double array above.

**Comment by gmponos** [comment]
Related to this: https://github.com/florianv/exchanger/pull/71

**Comment by Nyholm** [comment]
We are wrong to assume that `$candidates[0]['class']` is a string. It could be a callable as well. 

https://github.com/php-http/discovery/blob/master/src/Exception/NoCandidateFoundException.php#L30

It is the second bug on this class. We should add some easy to write tests in this repo as well.

**Comment by Nyholm** [comment]
I've fixed this issue here: #132
